src/trigger_completed.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2022 Collabora Limited
+# Author: Jeny Sadadia <jeny.sadadia@collabora.com>
+
+import logging
+import os
+import sys
+
+import kernelci
+import kernelci.config
+import kernelci.db
+from kernelci.cli import Args, Command, parse_opts
+
+from logger import Logger
+
+
+class TriggerCompleted:
+
+    def __init__(self, configs, args):
+        self._db_config = configs['db_configs'][args.db_config]
+        api_token = os.getenv('API_TOKEN')
+        self._db = kernelci.db.get_db(self._db_config, api_token)
+        self._logger = Logger("config/logger.conf", "trigger_completed")
+
+    def run(self):
+        sub_id = self._db.subscribe_node_channel(filters={
+            'op': ('created', 'updated'),
+            'status': ('completed', 'timeout'),
+        })
+        self._logger.log_message(logging.INFO, "Listening for node events... ")
+        self._logger.log_message(logging.INFO, "Press Ctrl-C to stop.")
+        sys.stdout.flush()
+
+        try:
+            while True:
+                node = self._db.receive_node(sub_id)
+                if node['parent'] is None:
+                    continue
+                resp = self._db.get_completed_event(node['parent'], seconds=60)
```

Comment:
See my other comment, we shouldn't be doing it this way. Passing a timeout in the API call and relying on the API to sleep is going to cause some big design issues. We should just be sleeping in the client, and then send a `POST` request to update nodes when the client has determined all the child nodes have completed.